src/mame/apple/apple2.cpp

```diff
@@ -937,6 +944,53 @@ void apple2_state::ay3600_data_ready_w(int state)
 	}
 }
 
+void apple2_state::tk10_data_write(u8 data)
+{
+		if (m_maincpu->total_cycles() < 25000)
+		{
+			return;
+		}
```

Comment:
This is definitely the wrong way to do this.  What’s actually going on here?  Is there a monostable that holds the microcontroller reset for a while after power-on?  If there is, is it also affected by reset?  Or is there some weird glue logic that just ignores data from the keyboard after power-on/reset?  Either way you should probably be using a timer.

Reply:
This has nothing to do with hardware and everything to do with MAME.  The keypress to start the emulation past the info screen leaks into the emulated system, and that was causing issues with disks where you boot them and then press a key to skip a title screen or select a menu item or whatever.

Reply:
I don't disagree that it can be a handy shortcut, but I also spent over a week trying to reconcile reports that MAME skipped the initial screens of a bunch of random software for no apparent reason.

Reply:
I got fooled by it as well – I was hitting an arrow key to dismiss the message, which toggles a switch by default on an Intel dev system in MAME.  I was wondering why the switch wasn’t in its default state when I started with no .cfg file.

Reply:
My strategy was to match the routines as closely as possible to the existing code for processing the keyboard.